libsignal-service/src/account_manager.rs

```diff
@@ -49,13 +49,37 @@ impl<Service: PushService> AccountManager<Service> {
         }
     }
 
+    pub async fn request_sms_verification_code_with_data(
```

Comment:
Feel free to break interfaces. @gferon and us are the main consumers of the library, and we break the interface all the time.

Reply:
Good to know. Do you care about keeping this close to the java lib interface? If not, I'll split the method in two again.

Reply:
I like to keep the reference (as in: comments to where to find the respective Java methods), but if I have to pick between something Rusty and something 1:1 compatible, I'm 100% in camp Rust.

Reply:
so would be different methods more rusty?

Reply:
Both method names would give away the intent of the call: there can be a `request_captcha` method, and then you can alter the original method (`request_sms`) with an argument `captcha: Option<String>`, or would that not make sense?

Reply:
Not sure I understand what you mean: I think there should be either three methods per type (sms/voice) `request_sms_verification_code_with_captcha`, `request_sms_verification_code_with_challenge` and `request_sms_verification_code` or only the last one with two optional arguments ...

Reply:
removed the "with data" methods. Seemed to be the easiest option for whisperfish to use.